Development/nmos/settings.cpp

```diff
@@ -54,8 +55,14 @@ namespace nmos
                 }
             }
 
+            // if the "server_address" settings is presented, use it for the node's "api endpoints" and "href", and device's "controls href"
+            const auto server_address = experimental::get_server_address(settings);
+            if (!server_address.empty())
+            {
+                web::json::insert(settings, std::make_pair(nmos::fields::host_address, server_address));
+            }
             // if the "host_address" setting was omitted, use the first of the "host_addresses"
-            if (settings.has_field(nmos::fields::host_addresses))
+            else if (settings.has_field(nmos::fields::host_addresses))
```

Comment:
Finishing this thought... I think it's probably better to separate the use of `client_address` and `server_address` for binding sockets, from the use of `host_address(es)` for putting in HTTP headers and the JSON data model?

Reply:
In the case of a reverse proxy (e.g. for TLS termination) the server_address is an implementation detail, where the host_address(es) are the public IP addresses of the proxy.

Reply:
Yes, server_address is an implementation detail, where the host_address(es) are the public IP addresses of the proxy.